### PR TITLE
simplify handling of privileged ports setting

### PR DESCRIPTION
**Pull request by arvidn**
by untangling it from the port filter

### CODE REVIEW

File: src/torrent.cpp [hunk 1]
```diff
@@ -11246,9 +11246,31 @@ namespace {
 		peers_erased(st.erased);
 	}
 
+	// this call will disconnect any peers whose remote port is < 1024
+	void torrent::privileged_port_updated()
+	{
+		if (!m_peer_list) return;
+
+		port_filter ports;
+		ports.add_rule(0, 1024, port_filter::blocked);
```
Comment (HanabishiRecca) [inline src/torrent.cpp:11255]: Shouldn't it be `0, 1023`? Privileged ports are **below** 1024.

File: src/torrent.cpp [hunk 2]
```diff
@@ -11246,9 +11246,31 @@ namespace {
...
+
+		torrent_state st = get_peer_list_state();
+		std::vector<address> banned;
+		m_peer_list->apply_port_filter(ports, &st, banned);
+
+		if (alerts().should_post<peer_blocked_alert>())
+		{
+			for (auto const& addr : banned)
+				alerts().emplace_alert<peer_blocked_alert>(get_handle()
+					, tcp::endpoint(addr, 0)
```
Comment (HanabishiRecca) [inline src/torrent.cpp:11265]: Can we preserve the port number for the alert?
I am actually pushing https://github.com/qbittorrent/qBittorrent/pull/18544 atm.

Comment (arvidn) [reply src/torrent.cpp:11265]: yeah, that makes sense

Comment (HanabishiRecca) [reply src/torrent.cpp]: Thanks.

**Comment by arvidn** [comment]
@HanabishiRecca does this patch fix the issue you experienced?

**Comment by HanabishiRecca** [comment]
Yes.

image

Will it be backported to 1.x?

**Comment by HanabishiRecca** [comment]
I have a tangential question: https://github.com/arvidn/libtorrent/issues/7315

**Pull request merged**